### PR TITLE
santactl fileinfo updates

### PR DESCRIPTION
**Pull request by tburgin**
*  	Added --json output option. Added --key output option.
*  	Added multi-file processing.

### CODE REVIEW

File: Source/santactl/Commands/SNTCommandFileInfo.m [hunk 1]
```diff
@@ -43,190 +366,221 @@ + (NSString *)shortHelpText {
...
+      return;
+    } else {
+      [paths addObject:args[idx]];
     }
+  }];
+  if (*key && (!*certIndex) && (![self.fileInfoKeys containsObject:*key])) {
+    [self printErrorUsageAndExit:
+        [NSString stringWithFormat:@"\n\"%@\" is an invalid key", *key]];
   }
+  if (*key && (*certIndex) && (![self.signingChainKeys containsObject:*key])) {
```
Comment (russellhancox) [inline Source/santactl/Commands/SNTCommandFileInfo.m]: else if

File: Source/santactl/Commands/SNTCommandFileInfo.m [hunk 2]
```diff
@@ -43,190 +366,221 @@ + (NSString *)shortHelpText {
...
+          SNTAttributeBlock block = fi.propertyMap[key];
+          outputHash[key] = block(fi);
+        }];
+      }
+      [outputHashes addObject:outputHash];
+    }];
+    hashOperation.qualityOfService = NSQualityOfServiceUserInitiated;
+    hashOperation.completionBlock = ^{
+      if (isatty(STDOUT_FILENO) && !jsonOutput && filePaths.count > 1) {
+        printf("\r Hashing %lu/%lu", ++hashed, filePaths.count);
```
Comment (russellhancox) [inline Source/santactl/Commands/SNTCommandFileInfo.m]: s/Hashing/Calculating/

File: Source/santactl/Commands/SNTCommandFileInfo.m [hunk 3]
```diff
@@ -43,190 +366,221 @@ + (NSString *)shortHelpText {
...
-    [self printKey:@"Download Referer URL" value:fileInfo.quarantineRefererURL];
-    [self printKey:@"Download URL" value:fileInfo.quarantineDataURL];
-    [self printKey:@"Download Timestamp"
-             value:[dateFormatter stringFromDate:fileInfo.quarantineTimestamp]];
-    [self printKey:@"Download Agent" value:fileInfo.quarantineAgentBundleID];
-  }
+      if (key) {
+        if (certIndex) {
+          SNTAttributeBlock block = fi.signingChain;
+          [block(fi) enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
```
Comment (russellhancox) [inline Source/santactl/Commands/SNTCommandFileInfo.m]: I'd re-write this as:

```objc
NSArray *signingChain = fi.signingChain(fi);
[signingChain enumerateObjectsUsingBlock:...]
```

File: Source/santactl/Commands/SNTCommandFileInfo.m [hunk 4]
```diff
@@ -43,190 +366,221 @@ + (NSString *)shortHelpText {
...
+             @"           valid Keys:\n"
+             @"%@\n"
+             @"           valid keys when using --cert-index:\n"
+             @"%@\n"
+             @"    --cert-index: an integer corresponding to a certificate of the signing chain"
+             @"\n"
+             @"Example: santactl fileinfo --cert-index 0 --key SHA-256 --json /usr/bin/yes\n"
+             @"         santactl fileinfo --key SHA-256 --json /usr/bin/yes\n"
+             @"         santactl fileinfo /usr/bin/yes /bin/*\n"
+             , [self printKeyArray:[self fileInfoKeys]],
```
Comment (russellhancox) [inline Source/santactl/Commands/SNTCommandFileInfo.m]: Move first comma to end of previous line

File: Source/santactl/Commands/SNTCommandFileInfo.m [hunk 5]
```diff
@@ -43,190 +366,221 @@ + (NSString *)shortHelpText {
...
+                                                       options:NSJSONWritingPrettyPrinted
+                                                         error:NULL];
+    printf("%s\n", [[NSString alloc] initWithData:jsonData
+                                         encoding:NSUTF8StringEncoding].UTF8String);
+    return;
   }
 
-  exit(0);
+  [outputHashes enumerateObjectsUsingBlock:^(id outputHash, NSUInteger idx, BOOL *stop) {
+    if ([outputHash count] == 1) {
```
Comment (russellhancox) [inline Source/santactl/Commands/SNTCommandFileInfo.m]: outputHash.count

Comment (tburgin) [inline Source/santactl/Commands/SNTCommandFileInfo.m]: outputHash is of type `id`. I could cast outputHash or just leave accessing the property with the getter method.

**Pull request merged**